src/examples/libpmem2/map_multiple_files/map_multiple_files.c

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * map_multiple_files.c -- implementation of virtual address allocation example
+ */
+
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <io.h>
+#endif
+#include <libpmem2.h>
+
+/*
+ * file_dsc - a structure that keeps information about file mapping
+ */
+struct file_dsc {
+	int fd;
+	size_t size;
+	struct pmem2_source *src;
+	struct pmem2_map *map;
+};
+
+/*
+ * file_dsc_init - initialize file_dsc structure values
+ */
+static int
+file_dsc_init(struct file_dsc *fdsc, char *path)
+{
+	if ((fdsc->fd = open(path, O_RDWR)) < 0) {
+		perror("open");
+		goto fail;
+	}
+
+	if (pmem2_source_from_fd(&fdsc->src, fdsc->fd)) {
+		pmem2_perror("pmem2_source_from_fd");
+		goto file_close;
+	}
+
+	if (pmem2_source_size(fdsc->src, &fdsc->size)) {
+		pmem2_perror("pmem2_source_size");
+		goto source_delete;
+	}
+
+	return 0;
+
+source_delete:
+	pmem2_source_delete(&fdsc->src);
+file_close:
+	close(fdsc->fd);
+fail:
+	return 1;
+}
+
+/*
+ * file_dsc_fini - deinitialize file_dsc structure values
+ */
+static void
+file_dsc_fini(struct file_dsc *fdsc)
+{
+	close(fdsc->fd);
+	pmem2_source_delete(&fdsc->src);
+}
+
+/*
+ * file_check_align - check if file is aligned
+ */
+static int
+file_check_align(struct file_dsc *fdsc)
+{
+	size_t alignment;
+	pmem2_source_alignment(fdsc->src, &alignment);
+
+	if (fdsc->size % alignment != 0) {
+		fprintf(stderr,
+			"usage: files must be aligned\n");
+		return 1;
+	}
+
+	return 0;
+}
+
+/*
+ * files_check_same_align - check if files have the same alignment
+ */
+static int
+files_check_same_align(struct file_dsc **fdsc, size_t nfiles)
+{
+	size_t alignment, nalignment;
+	pmem2_source_alignment((*fdsc)[0].src, &alignment);
+
+	for (int n = 1; n < nfiles; n++) {
+		pmem2_source_alignment((*fdsc)[n].src, &nalignment);
+		if (alignment != nalignment) {
+			fprintf(stderr,
+				"usage: files must have the same alignment\n");
+			return 1;
+		}
+	}
+
+	return 0;
+}
+
+/*
+ * files_check_memset - check if mappings retrieve the same memset function
+ */
+static int
+files_check_same_memset(struct file_dsc *fdsc, size_t nfiles,
+		pmem2_memset_fn memset_fn)
+{
+	for (int n = 0; n < nfiles; n++) {
```

Comment:
nfiles is size_t and you are comparing it to int, which a) have different size b) have different signedness

please fix all instances of this bug